File: lmcc/lmcc-lang/tests/fixtures/structure.cc
namespace sample {
class Runner {
public:
    int run(int value) {
        auto adjust = [value](int step) {
            for (int item : {1, 2}) {
                if (item > 0) {
                    switch (item) {
                        case 1: value += step; break;
                        default: break;
                    }
                }
            }
            return value;
        };
        try {
            return adjust(1);
        } catch (...) {
            return 0;
        }
    }
};

enum class State { ready, done };
}

File: testdata/lang/structure.py
class Structure:
    def run(self, value):
        adjust = lambda step: value + step
        for item in (1, 2):
            if item > 0:
                match item:
                    case 1:
                        value += 1
                    case _:
                        value += 0
        try:
            while value < 4:
                value += 1
        except ValueError:
            return 0
        finally:
            value += 1
        return adjust(1)
